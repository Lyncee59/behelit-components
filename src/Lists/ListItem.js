import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { theme } from '../Tools/interpolation'
import { Text } from '../Typography'

const Wrapper = styled.li`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-start;
`
const BulletPoint = styled.div`
  display: flex;
  flex-shrink: 0;
  width: 5px;
  height: 5px;
  border: 2px solid ${theme('listItemColor')};
  border-radius: 50%;
  margin-right: 10px;
  margin-top: 5px;
`

const ListItem = ({ children, size, weight, ...rest }) => (
  <Wrapper {...rest}>
    <BulletPoint />
    <Text size="1rem" weight={400}>{children}</Text>
  </Wrapper>
)

ListItem.propTypes = {
  size: PropTypes.string,
  weight: PropTypes.number
}

ListItem.defaultProps = {
  size: '1rem',
  weight: 400
}

export default ListItem
