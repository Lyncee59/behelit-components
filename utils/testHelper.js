import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/Theme'

export const mountWithTheme = (tree) => mount(<ThemeProvider>{tree}</ThemeProvider>)