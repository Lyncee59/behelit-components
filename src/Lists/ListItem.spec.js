import React from 'react'
import { mountWithTheme } from '../../utils/testHelper'

import ListItem from './ListItem'

describe('ListItem', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<ListItem>List</ListItem>)

    expect(tree.find(ListItem)).toMatchSnapshot()
  })
})
