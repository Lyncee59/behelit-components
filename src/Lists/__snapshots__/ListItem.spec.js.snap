// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render correctly 1`] = `
.c2 {
  font-family: 'Roboto',sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  color: #323232;
  opacity: 1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 5px;
  height: 5px;
  border: 2px solid #990000;
  border-radius: 50%;
  margin-right: 10px;
  margin-top: 5px;
}

<ListItem
  size="1rem"
  weight={400}
>
  <styled.li>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-start;
",
            ],
          },
          "displayName": "styled.li",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "li",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <li
        className="c0"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  display: flex;
  flex-shrink: 0;
  width: 5px;
  height: 5px;
  border: 2px solid ",
                    [Function],
                    ";
  border-radius: 50%;
  margin-right: 10px;
  margin-top: 5px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c1"
            />
          </StyledComponent>
        </styled.div>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <StyledComponent
              capitalize={false}
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "c2",
                    "rules": Array [
                      "
  font-family: ",
                      [Function],
                      ";
  font-weight: ",
                      [Function],
                      ";
  font-size: ",
                      [Function],
                      ";
  text-transform: ",
                      [Function],
                      ";
  font-style: ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  opacity: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              italic={false}
              opacity={1}
              size="1rem"
              uppercase={false}
              weight={400}
            >
              <span
                className="c2"
                opacity={1}
                size="1rem"
              >
                List
              </span>
            </StyledComponent>
          </styled.span>
        </Text>
      </li>
    </StyledComponent>
  </styled.li>
</ListItem>
`;
